autoscale/src/main/java/com/sequenceiq/periscope/init/AutoscaleStartupService.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.periscope.init;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import com.sequenceiq.ambari.client.AmbariClient;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.MetricAlert;
+import com.sequenceiq.periscope.repository.ClusterRepository;
+import com.sequenceiq.periscope.service.AmbariClientProvider;
+
+@Component
+public class AutoscaleStartupService implements ApplicationListener<ContextRefreshedEvent> {
+    @Inject
+    private ClusterRepository clusterRepository;
+
+    @Inject
+    private AmbariClientProvider ambariClientProvider;
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        Iterable<Cluster> clusters = clusterRepository.findAll();
+        Set<Cluster> changedClusters = new HashSet<>();
+        for (Cluster cluster : clusters) {
+            if (isMetricAlertMigrationNeeded(cluster)) {
```

Comment:
This could be included in the WHERE part of the SQL statement, instead of querying all of the clusters.